tokio/src/runtime/scheduler/current_thread.rs

```diff
@@ -569,6 +569,8 @@ impl CoreGuard<'_> {
                         }
                     };
 
+                    let task_id = task.get_task_id();
+                    let _task_id_guard = TaskIdGuard::new(task_id);
                     let task = context.handle.shared.owned.assert_owner(task);
```

Comment:
Why is this being set in the current-thread runtime. It should happen somewhere in the harness code instead.

Reply:
Why should it be set in the harness code? Don't know the harness code that well, but doesn't this just implement the polling and waking functionality of tasks?

Seems fairly reasonable to me to set the task id before running a task.

Reply:
It's important that we always remember to set the task id, and we shouldn't do it more than once. Setting it in the harness code, which is also responsible for e.g. catching panics when polling the task, makes sure that we have to do it only in one place while making sure that we can't forget.

For example, in this PR, you set it in the current-thread runtime, but _not_ in the multi-thread runtime. However, you also set it in the `run` function, which is called for both current-thread and multi-thread, so you're actually setting it twice for current-thread.

Reply:
Makes sense, I'll try to change this later today.